catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -271,9 +269,24 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
       throw new UnsupportedQueryException("Could not complete solr query.", e);
     }
 
+    responseProps.put("query", query.get("q"));
     return new SourceResponseImpl(request, responseProps, results, totalHits);
   }
 
+  private boolean userSpellcheckIsOn(QueryRequest request) {
+    Boolean userSpellcheckChoice = false;
+    if (request.getProperties().get("spellcheck") != null) {
+      userSpellcheckChoice = (Boolean) request.getProperties().get("spellcheck");
+    }
+    return userSpellcheckChoice;
+  }
+
+  private boolean solrSpellcheckHasResults(QueryResponse solrResponse) {
+    return solrResponse.getSpellCheckResponse() != null
+        && solrResponse.getResults().size() == 0
```

Comment:
can getResults be null?

Reply:
No because if there are no results returned from the query when `clientquery(query, METHOD.POST)` is called like in line 220, solrResponse.getResults() is just initialized as an empty list. 
You can see in line 223  that the same thing happens and then the results get looped through.